frontend/app/work_packages/controllers/work-package-new-controller.js

```diff
@@ -0,0 +1,172 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function(
+           $scope,
+           $rootScope,
+           $state,
+           $stateParams,
+           $timeout,
+           $window,
+           PathHelper,
+           WorkPackagesOverviewService,
+           WorkPackageFieldService,
+           WorkPackageService,
+           EditableFieldsState
+           ) {
+
+  var vm = this,
+      unhideableFields = [
+        'subject',
+        'type',
+        'status',
+        'description',
+        'priority',
+        'assignee',
+        'percentageDone'
+      ];
+
+  vm.groupedFields = [];
+  vm.hideEmptyFields = true;
+
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  vm.loaderPromise = null;
+
+  vm.isGroupHideable = isGroupHideable;
+  vm.isFieldHideable = isFieldHideable;
+  vm.getLabel = getLabel;
+  vm.isSpecified = isSpecified;
+  vm.isEditable = isEditable;
+  vm.hasNiceStar = hasNiceStar;
+  vm.showToggleButton = showToggleButton;
+
+  activate();
+
+  function activate() {
+    EditableFieldsState.forcedEditState = true;
+    vm.loaderPromise = WorkPackageService.initializeWorkPackage($scope.projectIdentifier, {
+      type: PathHelper.apiV3TypePath($stateParams.type)
+    }).then(function(wp) {
+      vm.workPackage = wp;
+      $scope.workPackage = wp;
+      var firstTimeFocused = false;
+      $scope.$watchCollection('vm.workPackage.form', function() {
+        vm.groupedFields = WorkPackagesOverviewService.getGroupedWorkPackageOverviewAttributes();
+        var schema = WorkPackageFieldService.getSchema(vm.workPackage);
+        var otherGroup = _.find(vm.groupedFields, { groupName: 'other' });
+        otherGroup.attributes = [];
+        _.forEach(schema.props, function(prop, propName) {
+          if (propName.match(/^customField/)) {
+            otherGroup.attributes.push(propName);
+          }
+        });
+        otherGroup.attributes.sort(function(a, b) {
+          return getLabel(a).toLowerCase().localeCompare(getLabel(b).toLowerCase());
+        });
+        if (!firstTimeFocused) {
+          firstTimeFocused = true;
+          $timeout(function() {
+            // TODO: figure out a better way to fix the wp table columns bug
+            // where arrows are misplaced when not resizing the window
+            angular.element($window).trigger('resize');
+            angular.element('.work-packages--details--subject .focus-input').focus();
+          });
+        }
```

Comment:
If this whole `firstTimeFocused` business is because of a bug in the WP table, why do I not see it in the otherwise identical code for the edit mode? ([identical code here](https://github.com/NobodysNightmare/openproject/blob/7484c1980a57485552215f823364d238f1df9665/frontend/app/work_packages/controllers/details-tab-overview-controller.js#L60-73))

And by extension: Why is this code duplicated?